bookOverview/src/main/kotlin/voice/bookOverview/editBookCategory/EditBookCategoryViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package voice.bookOverview.editBookCategory
+
+import com.squareup.anvil.annotations.ContributesMultibinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import voice.bookOverview.bottomSheet.BottomSheetItem
+import voice.bookOverview.bottomSheet.BottomSheetItemViewModel
+import voice.bookOverview.di.BookOverviewScope
+import voice.bookOverview.overview.BookOverviewCategory
+import voice.bookOverview.overview.category
+import voice.data.Book
+import voice.data.repo.BookRepository
+import javax.inject.Inject
+
+@BookOverviewScope
+@ContributesMultibinding(BookOverviewScope::class)
+class EditBookCategoryViewModel
+@Inject
+constructor(
+  private val repo: BookRepository,
+) : BottomSheetItemViewModel {
+
+  private val scope = CoroutineScope(Dispatchers.IO)
```

Comment:
Reason why I wanted `BottomSheetItemViewModel.onItemClicked` to be a suspend function is to avoid creating a scope in the ViewModel. :smiley:

Now that ViewModels inherit from `androidx.lifecycle.ViewModel`, would you like to use [viewModelScope](https://developer.android.com/topic/libraries/architecture/coroutines#lifecycle-aware) instead?

Reply:
That would not be a good choice as that scope is cancelled when the view model is cleared.
However after a user pressed on that button that action should completed nevertheless - regardless of the user leaving the screen at the time or not.

Reply:
Yes. my main concern is using all these `MainScope` instances in VMs without a `MainScope.cancel()` as you normally would have [if you used it in an activity](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-main-scope.html)